### ISSUE TITLE
Enable consensus encoding to a `fmt::Formatter`

### ISSUE THREAD

**Issue by tcharding** [rust-bitcoin/rust-bitcoin#2513]
Currently one cannot use `consensus_encode` and pass it a `fmt::Formatter` - one would be able to do so if we used `std::io::Write` instead of our own `io::Write` trait therefore it seems reasonable to expect this functionality.

**Comment by Kixunil** [comment]
I think it'd be even better to have a bridge for `fmt::Write` outputting hex. One could then also use my crate `fmt2io` to write hex to `io::Write`.

**Comment by tcharding** [comment]
> I don't believe this has ever been possible. core::fmt::Write and std::io::Write are completely different traits.

You are correct, I saw `fmt::Formatter` implements `Write` and confused it with `core::io::Write`.

**Comment by tcharding** [comment]
The old encoding traits are destined for the graveyard eventually now that we have the `consensus_encoding` crate, closing this.
